pandera/schemas.py

```diff
@@ -689,6 +689,168 @@ def to_yaml(self, fp: Union[str, Path] = None):
 
         return pandera.io.to_yaml(self, fp)
 
+    def set_index(
+        self, keys: List[str], drop: bool = True, append: bool = False
+    ):
+        """
+        A method for setting the :class:`Index` of a :class:`DataFrameSchema`,
+        via an existing :class:`Column` or list of :class:`Column`s.
+
+        :param keys: list of labels
+        :param drop: bool, default True
+        :param append: bool, default False
+        :return: a new :class:`DataFrameSchema` with specified column(s) in the index.
+        """
+        from pandera.schema_components import Index, MultiIndex
+
+        new_schema = copy.deepcopy(self)
+
+        keys_temp: List = list(keys) if not isinstance(keys, List) else keys
+
+        # ensure all specified keys are present in the columns
+        try:
+            not_in_cols: List[str] = [
+                x for x in keys_temp if x not in new_schema.columns.keys()
+            ]
+            assert not_in_cols == []
+        except AssertionError:
+            raise Exception(f"Keys {not_in_cols} not found in schema columns!")
```

Comment:
these exceptions should be `errors.SchemaInitError` and use the `from` keyword so that full error trace is available. `assert` statements are also nice when quickly scripting things, but in this case it might be more appropriate to use a conditional then `raise` here.

```python
not_in_cols: List[str] = [
    x for x in keys_temp if x not in new_schema.columns.keys()
]
if not not_in_cols:
     raise errors.SchemaInitError(f"Keys {not_in_cols} not found in schema columns!")
```